### PR TITLE
Update selectmenu explainer

### PR DESCRIPTION
**Pull request by josepharhar**
This patch adds a background paragraph to help explain the motivation behind the selectmenu element and a link to open issues.

### CODE REVIEW

**Review by andrico1234 — APPROVED** [review]

File: research/src/pages/prototypes/selectmenu.mdx
Comment (andrico1234) [inline research/src/pages/prototypes/selectmenu.mdx:14]: "which leads to them _implementing_ their"

Comment (andrico1234) [reply research/src/pages/prototypes/selectmenu.mdx:14]: Can we break up the two sentences?

The `<select>` element does not provide enough customization for web developers, which leads them to implementing their own. These implementations can lead to reduced performance, reliability, and accessibility over the native form control elements.

**Review by andrico1234 — CHANGES_REQUESTED** [review]
Thanks for raising this. I've added some comments relating to typos and cleaning up the sentences.

**Comment by josepharhar** [comment]
> After reviewing further. I'm not sure if this file is the right place for an introduction to the selectmenu component. The file is strictly related to the selectmenu prototype and links to the proposal in the introduction.

Should I create a new file for the explainer? https://open-ui.org/components/select seems like it has a better location but it only seems to describe the existing `<select>` element rather than the new `<selectmenu>` element. This file seems to have everything I want in it maybe except for the "How to use the `<selectmenu>` component" and "Help test the control" headings. If I created a new file then I would copy most of the stuff out of this file.

**Comment by andrico1234** [comment]
Good point, it might be worth reaching out in the Discord to some of the folks that have worked on the selectmenu proposal

**Comment by josepharhar** [comment]
I asked on discord and didn't get a response.
@gregwhitworth @dandclark any thoughts?

**Comment by josepharhar** [comment]
I talked to @mfreed7 and it sounds like maybe we should just move this file to an explainer-appropriate location, like wherever the popover explainer is: https://open-ui.org/components/popover.research.explainer

How do I do that? Should something be left in the current location to redirect it?

**Comment by josepharhar** [comment]
Based on the discussion in discord, we should make the selectmenu explainer go in components/selectmenu.
I moved most of the contents of prototypes/selectmenu.mdx to components/selectmenu.mdx and made a link from the prototypes one to the components one.

**Comment by andrico1234** [comment]
Thanks for making these changes!

I've got a big PR request waiting to go in. Are you ok for me to merge this in after? I hope to merge it in over the next week. (if not, feel free to ping me and i'll merge this :) )

File: research/src/pages/prototypes/selectmenu.mdx
Comment (josepharhar) [reply research/src/pages/prototypes/selectmenu.mdx]: done

**Comment by josepharhar** [comment]
I resolved the conflicts. Would you mind checking out the new file I added to make sure it conforms to the new website?

File: research/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline research/src/pages/components/selectmenu.mdx:4]: @josepharhar , you'll need to import the layout here

File: research/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline research/src/pages/components/selectmenu.mdx:3]: i don't think this is necessary since we automatically build the path using the file directory

File: research/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline research/src/pages/components/selectmenu.mdx:4]: I think you can remove this, since it's `true` by default

Comment (andrico1234) [reply research/src/pages/components/selectmenu.mdx:4]: it'll look something like this:
`layout: ../../layouts/ComponentLayout.astro`

File: research/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline research/src/pages/components/selectmenu.mdx:1]: You'll also need to add the menu section you want this displayed under:

```
menu: Research
```

**Comment by gregwhitworth** [comment]
Based on the discussion in discord, we should make the selectmenu explainer go in components/selectmenu.
I moved most of the contents of prototypes/selectmenu.mdx to components/selectmenu.mdx and made a link from the prototypes one to the components one.

Yes, we should move this to selectmenu (Explainer). One key caveat however; this is relatively high-level when compared with Popover. There have been numerous discussions around `<selectmenu>` and resolutions that need to be captured. However; I know that you all are actively working on a spec variant in WHATWG. Because of this we have two options:

1. We have the specific implementation details (akin to what's in `<select>` now)
2. We link out to the PR for specific implementation details

I don't want to cause more work as the current content under `<select>` is outdated to the work occurring in Open UI. I agree with the other requests made by @andrico1234

**Comment by gregwhitworth** [comment]
Also, @josepharhar can you please make sure this content is captured in either location: https://github.com/openui/open-ui/pull/441

cc: @scottaohara

**Comment by mfreed7** [comment]
> One key caveat however; this is relatively high-level when compared with Popover. There have been numerous discussions around `<selectmenu>` and resolutions that need to be captured. However; I know that you all are actively working on a spec variant in WHATWG. Because of this we have two options:

I really think we need to have an explainer that is on-par with the popover explainer, in terms of detail. There should be one place to go for folks wanting to come up to speed on the feature, and that includes the technical detail. So I'd prefer that, vs. linking out to spec text, which isn't very user-friendly in the first place.

Having said that, I really think this PR should land soon, and then we can tackle improving the detail in followon PRs. The current "explainer" is confusing at best since it doesn't really explain anything close to what we're building.

File: research/src/pages/components/selectmenu.mdx
Comment (josepharhar) [reply research/src/pages/components/selectmenu.mdx:4]: done

File: research/src/pages/components/selectmenu.mdx
Comment (josepharhar) [reply research/src/pages/components/selectmenu.mdx:3]: done

File: research/src/pages/components/selectmenu.mdx
Comment (josepharhar) [reply research/src/pages/components/selectmenu.mdx:1]: I did "proposals" since I want it to show up in the proposals section in the sidebar

**Comment by andrico1234** [comment]
Thanks for updating @josepharhar . The page displays in the navbar, and the content renders correctly!

I noticed that the images aren't being displayed anymore though.

image

Reach out if you have any issues with this :)

**Comment by josepharhar** [comment]
I agree with mason that having a more detailed explainer is better because I don't want to start an HTML spec PR until it seems like things aren't changing much anymore. It is a lot of work to keep an HTML spec PR up to date when things are changing, and I'd like to introduce this to the HTML people with a detailed explainer rather than an actual PR. I will work on followup PRs to add detail to this explainer.

**Comment by josepharhar** [comment]
@andrico1234 how do I build/test this PR? I haven't actually tried yet lol

File: site/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline site/src/pages/components/selectmenu.mdx:8]: Don't forget the `astro` extension

import Image from '../../components/image.astro'

File: site/src/pages/components/selectmenu.mdx
Comment (andrico1234) [inline site/src/pages/components/selectmenu.mdx:251]: I've noticed that the other images use the following prefix before the file name "/images/". did you intentionally omit this?

File: site/src/pages/components/selectmenu.mdx
Comment (josepharhar) [reply site/src/pages/components/selectmenu.mdx:8]: done

**Comment by andrico1234** [comment]
@josepharhar , at the bottom here you'll see a section called "netlify/open-ui/deploy-preview"

image

If you click on the "Details" it takes you directly to a preview version of the site

File: site/src/pages/components/selectmenu.mdx
Comment (josepharhar) [reply site/src/pages/components/selectmenu.mdx]: This was not intentional, I've added the absolute path

**Comment by andrico1234** [comment]
That did the trick 😎

image

I see that there are still two instances of missing images though: 

They have the alt text: "The rendering of a selectmenu control with the above HTML and CSS code"

**Comment by josepharhar** [comment]
I added more `/image/` prefixes and it looks like it's all working now

**Review by andrico1234 — APPROVED** [review]

**Pull request merged**
